idmapping-common/src/main/java/org/uniprot/api/idmapping/common/repository/UniprotKBMappingRepository.java

```diff
@@ -44,8 +44,12 @@ public UniProtKBEntry getDeletedEntry(String accession) {
                     if (Utils.notNullNotEmpty(document.inactiveReason)) {
                         String[] inactiveReason = document.inactiveReason.split(":");
                         if (inactiveReason.length == 2) {
-                            inactiveReasonBuilder.deletedReason(
-                                    DeletedReason.valueOf(inactiveReason[1].strip()));
+                            if (inactiveReason[0].equals(InactiveReasonType.DELETED.toString())) {
+                                inactiveReasonBuilder.deletedReason(
+                                        DeletedReason.valueOf(inactiveReason[1].strip()));
+                            } else {
+                                return getDeletedEntry(inactiveReason[1].strip());
+                            }
                         }
```

Comment:
It looks okay but if we could add code to handle demerged case "DEMERGED:P0A0K9,P0A0L0,P0C1S7,Q2FYS5". it would great.